src/proc/color-formats-converter.cpp

```diff
@@ -354,6 +358,207 @@ namespace librealsense
+                    r[14], g[14], b[14], 255, r[15], g[15], b[15], 255,
+                };
+                librealsense::copy(*dst, out, sizeof out);
+                *dst += sizeof out;
+                continue;
+            }
+
+            if (FORMAT == RS2_FORMAT_BGRA8)
+            {
+                uint8_t out[16 * 4] = {
+                    b[0], g[0], r[0], 255, b[1], g[1], r[1], 255,
+                    b[2], g[2], r[2], 255, b[3], g[3], r[3], 255,
+                    b[4], g[4], r[4], 255, b[5], g[5], r[5], 255,
+                    b[6], g[6], r[6], 255, b[7], g[7], r[7], 255,
+                    b[8], g[8], r[8], 255, b[9], g[9], r[9], 255,
+                    b[10], g[10], r[10], 255, b[11], g[11], r[11], 255,
+                    b[12], g[12], r[12], 255, b[13], g[13], r[13], 255,
+                    b[14], g[14], r[14], 255, b[15], g[15], r[15], 255,
+                };
+                librealsense::copy(*dst, out, sizeof out);
+                *dst += sizeof out;
+                continue;
+            }
+
+            
+        }
+    }
+
+    /////////////////////////////
+    // NV12 unpacking routines //
+    /////////////////////////////
+    // This templated function unpacks NV12i into Y8/Y16/RGB8/RGBA8/BGR8/BGRA8, depending on the compile-time parameter FORMAT.
+    // It is expected that all branching outside of the loop control variable will be removed due to constant-folding.
+    // This NV12i format is an INTERNAL FORMAT. Its configuration is as following: 2 lines of Y then one line of UV (line size is width)
+    // There is one Y value for each pixel, and one pair of U,V values for 4 pixels. 
+    // Their matching is similar to NV12 format - see: https://www.kernel.org/doc/html/v4.9/media/uapi/v4l/pixfmt-nv12.html?highlight=nv12
+    // For example: for the first 3 lines of the frame:
+    // Y0  Y1   Y2   Y3   ....
+    // Yw  Yw+1 Yw+2 Yw+3 .... (Yw:Ywidth)
+    // U0  V0   U1   V1
+    // The first pixel is (Y0, U0, V0), second pixel is (Y1, U0, V0)
+    // The first pixel in the second line is (Yw, U0, V0) second pixel in second line is (Yw+1, U0, V0)
+    // The third pixel in second line is (Yw+2, U1, V1)
+    template<rs2_format FORMAT> void unpack_nv12i(byte* const d[], const byte* s, int width, int height, int actual_size)
+    {
+        auto n = width * height;
+        assert(n % 16 == 0); // All currently supported color resolutions are multiples of 16 pixels. Could easily extend support to other resolutions by copying final n<16 pixels into a zero-padded buffer and recursively calling self for final iteration.
+
+        auto src = reinterpret_cast<const uint8_t*>(s);
+        auto dst = reinterpret_cast<uint8_t*>(d[0]);
+
+        auto src_height = height * 12 >> 3;
+
+        if (FORMAT == RS2_FORMAT_Y8)
+        {
+            // TODO REMI - check height is correct here (shall be equal to 1.5 * height - for height = 480, shall be 720)
```

Comment:
What can you check here? 
Need to remove the comment unless you really have something to check

Reply:
Done